src/compressor.rs

```diff
@@ -207,6 +212,18 @@ pub(crate) fn compress(
             }
             Ok(buf)
         },
+        #[cfg(feature = "zstd")]
+        (Compressor::Zstd, option) => {
+            let compression_level = match option {
+                None => 3,
+                Some(CompressionOptions::Zstd(option)) => option.compression_level,
+                Some(_) => unreachable!(),
```

Comment:
Before this is merged, I want to know about this `unreachable!` and `unwrap`.

Is unclear to me if the `Compressor`/`CompressorOption` always matches, and should be validated prior to this function or at this function. I assume this should be validated at the super-block read, so that's why the `unreachable!`.

About the `unwrap`, the `new` function will never fail, unless the `compression_level` is invalid `-7<x || x>22`, what perhaps should be verified on the `CompressorOption` creation, and not at every new compression call.

Reply:
BTW `unwrap` on line 222. The unwrap on line `127` never happens.

Reply:
They should match yes, especially if it's derived from an image as it's read in from here and only reads the matching options: https://github.com/wcampbell0x2a/backhand/blob/093ccc25abb577298f918e0ae1a553ae04f4f071/src/squashfs.rs#L296

However, you could add the wrong one or change only the compression type once you have a `FilesystemWriter`. So yes it should be verified here and made un-branchable here for perf. I think if it's asserted somewhere else it won't branch here?

I'll probably just make those fields private, so that we know that both are valid when changed.

I also want to expand to give `ExtraCompressionOptions` or something like that. Since since options like levels for Xz aren't stored in the image, and would be given to `FilesystemWrite`.

Also, we generate the encoder types at every compress, probably should init and do it once

Reply:
And technically, we could just store the `CompressionOptions`, since at this point we don't need to store the identifier value enum